packages/ui-app/src/Params/Param/Balance.tsx

```diff
@@ -0,0 +1,192 @@
+        newValue = new BN(inputConvertedFromScientificNotation);
+      } else {
+        newValue = new BN(value || '0');
+      }
+
+      onChange({
+        isValid,
+        value: newValue
+      });
+    } catch (e) {
+      console.error('error: ', e);
+    }
+  }
+
+  onKeyDown = (event: any): void => {
+    const { t } = this.props;
+
+    const altKey = 18;
+    const arrowLeftKey = 37;
+    const arrowRightKey = 39;
+    const backspaceKey = 46;
+    const cmdKey = event.metaKey; // firefox (91), safari (224)
+    const ctrlKey = 17;
+    const decimalPointKey = 190; // decimals allowed for scientific or exponential notation
+    const delKey = 8;
+    const escapeKey = 27;
+    const enterKey = 13;
+    const plusKey = 187; // '+'
+    const tabKey = 9; // next input field
+    const aKey = 65; // select all from balance input
+    const xKey = 88; // cut balance
+    const cKey = 67; // copy balance
+    const eKey = 69; // scientific or exponential notation
+    const vKey = 86; // paste balance
+    const zeroKey = 48;
+    const zeroNumpadKey = 96;
+
+    const regexE = /[e]/gi;
+    const regexPlus = /[\+]/gi;
+    const regexDecimalPoint = /[\.]/gi;
+
+    // only allow user balance input to contain one instance of 'e', '+', and '.' for decimal in scientific or exponential notation
+    if (
+      (event.keyCode === eKey && event.target.value.match(regexE)) ||
+      (event.keyCode === plusKey && event.target.value.match(regexPlus)) ||
+      (event.keyCode === decimalPointKey && event.target.value.match(regexDecimalPoint))
+    ) {
+      event.preventDefault();
+      return;
+    }
+
+    // obtain the current cursor index position in the Balance input field
+    const inputCursorIndex = event.target.value.slice(0, event.target.selectionStart).length;
+
+    // allow these keys
+    if ([escapeKey, enterKey, backspaceKey, delKey, tabKey, ctrlKey, altKey, cmdKey, eKey, arrowLeftKey,
+      arrowRightKey, decimalPointKey].includes(event.keyCode)) {
+      return;
+    }
+
+    // prevent user entering the '+' symbol unless immediately after the 'e' letter (i.e. 'e+') for exponential notation
+    if (event.keyCode === plusKey && event.target.value.charAt(inputCursorIndex - 1) !== 'e') {
+      event.preventDefault();
+      return;
+    }
+
+    // prevent user entering 0 at start index of input field
+    if ((event.keyCode === zeroKey || event.keyCode === zeroNumpadKey) && inputCursorIndex === 0) {
+      event.preventDefault();
+      return;
+    }
+
+    // allow users to to use cut/copy/paste combinations, but not non-numeric letters individually
+    // allow users to use the + key for exponential notation
+    if (
+      ((event.ctrlKey || cmdKey) && event.which === aKey) || // select all
+      ((event.ctrlKey || cmdKey) && event.which === cKey) || // copy
+      ((event.ctrlKey || cmdKey) && event.which === vKey) || // paste
+      ((event.ctrlKey || cmdKey) && event.which === xKey) || // cut
```

Comment:
`(event.ctrlKey || cmdKey)` is re-used, so something like this would work as well -

``(event.ctrlKey || cmdKey) && [aKey, cKey, vKey, xKey].includes(vKey)`

(Once again in the above example, just make sure we have the proper constant arrays defined.